### DISCUSSION TITLE
Javascript Cheat Sheet

### DISCUSSION THREAD

**Discussion by dudley-codes** [nss-day-cohort-47/C47-Discussions#28] [Look what I learned!]
I created a cheat sheet to help me get a better grasp on some of the stuff we've been learning and thought I'd share it with the group! Edit: I couldn't get this to post correctly as code, make sure you change the apostrophes out for back-ticks.

//  1. define an object
 const allObjects = [
    {
    itemOne: "string1",
    itemTwo: "string2",
    itemThree: "string3"
},
{
    itemOne: "string4",
    itemTwo: "string5",
    itemThree: "string6"
}
]

//  2. convert object to HTML representation
const oneIterationAsHTML = (object) => {
        return (
             ' <-----------------------------change this to a back-tick
            <div>${object.itemOne}<div>
            <div>${object.itemTwo}<div>
            <div>${object.itemThree}<div>
             '  <-----------------------------change this to a back-tick
   )
}

//  3. get reference to DOM element in which to put HTML representation
const DOMContainer = document.querySelector("#putHTMLClassHere")

//  4. Create for of loop to iterate through the data
let allAsHTML = ""

for(const oneObject of allObjects) {
    allAsHTML += oneIterationAsHTML(oneObject)
}

//  5. insert HTML representation into DOM element using innerHTML
DOMContainer.innerHTML += allAsHTML

**Comment by BryanNilsen** [comment]
This is a great start!

- wrap the code snippets with triple backticks to help the code stand out
- copy/paste the code from VSCode to retain the formatting too.

```
const allObjects = [
    {
        itemOne: "string1",
        itemTwo: "string2",
        itemThree: "string3"
    },
    {
        itemOne: "string4",
        itemTwo: "string5",
        itemThree: "string6"
    }
]
```
